services/motion/builtin/builtin.go

```diff
@@ -231,24 +236,111 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	// Shared variables for planning and execution threads, are all guarded with a mutex
+	var planMu sync.Mutex
+	var moveCtx context.Context
+	var currentPlan [][]referenceframe.Input = nil
+
+	// Shared variable between threads that can trigger replanning and planning thread
+	var replanFlag atomic.Bool
+	replanFlag.Store(true)
+
+	var successFlag atomic.Bool
+	successFlag.Store(false)
+	errChan := make(chan error)
+
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+
+		// maybe move this into the restart tickers function
+		cancelCtx, cancelFn := context.WithCancel(context.Background())
+		ms.cancelFn = cancelFn
+
+		positionPollingTicker := time.NewTicker(time.Duration(1000/motionCfg.PositionPollingFreqHz) * time.Millisecond)
+		obstaclePollingTicker := time.NewTicker(time.Duration(1000/motionCfg.ObstaclePollingFreqHz) * time.Millisecond)
+
+		for !successFlag.Load() && ctx.Err() == nil {
```

Comment:
Should this be cancelCtx? Or moveCtx?

Reply:
Again, I should have removed `moveCtx`, but forgot to so it shouldnt be that. I don't think it makes sense to have it be `cancelCtx` since this is the loop that we want to keep running and not exit out of even if cancelCtx is cancelled.  

Theres a chance I'm missing something though, what are your thoughts?

Reply:
What's the intended behavior of calling `ms.cancelFn()`?

As far as I can tell, currently the only thing it cancels is querying position from the movement sensor

Reply:
Ah, got it, thanks